namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -106,14 +108,42 @@ object HttpControlService {
       }
   }
 
+  trait NsPathUri {
+    val prefix: String
+
+    def unapply(uri: String): Option[(Ns, Path)] = {
+      if (uri.startsWith(prefix)) {
+        uri.stripPrefix(prefix).split("/").toSeq match {
+          case ns +: path if path.nonEmpty => Some((ns, Path.Utf8(path: _*)))
+          case _ => None
+        }
+      } else {
+        None
+      }
+    }
+  }
+
+  object BindUri extends NsPathUri {
+    val prefix = "/api/1/bind/"
+  }
+
+  object AddrUri extends NsPathUri {
+    val prefix = "/api/1/addr/"
+  }
+
+  object DelegateUri extends NsPathUri {
+    val prefix = "/api/1/delegate/"
+  }
+
```

Comment:
we already read namespaces from the URI eg `/api/1/dtabs/<ns>`.  Should we frontload the API churn for the sake of consistency and change these to `/api/1/dtabs?namespace=default` now?

Alternatively, we could have a transition period where we support both...